src/bin/main.rs

```diff
@@ -12,6 +12,12 @@ use yarner::util::PathUtil;
 use yarner::{templates, MultipleTransclusionError, ProjectCreationError};
 
 fn main() {
+    if let Err(err) = run() {
+        eprintln!("{}", err);
+    }
```

Comment:
@adamreichold Any better idea how to solve the problem that escape sequences in error messages are ignored when just returning `Result<...>` from main? If I don't "wrap" it into a function that prints the error, I get things like \n or \" instead of newline or quotes.

Reply:
You could create a custom error type and do any formatting in its `Debug` implementation to avoid this wrapping, but I would consider this a hack. If you do want control over how the errors are printed, I would say that this is the direct and also idiomatic way.  But note that you should still adjust you exit code here, i.e. use `fn main() -> i32` as the program should not return zero on failure. Otherwise, (CI) scripts will not be able to react to these errors.

(On a more general note, I think Yarner could have a much simpler code base if we would reconsider some of the design choices inherited from Outline, e.g. use simple boxed errors, drop the library/binary split, etc. I guess an overhaul or even a rewrite would be nice holiday project?)

Reply:
As returning `i32` is not possible, I would use this:

```rust
fn main() {
    std::process::exit(match run() {
        Ok(_) => 0,
        Err(err) => {
            eprintln!("{}", err);
            1
        }
    });
}
```

Reply:
I also think that a general review and rework of the code would make sense, although I would probably not do a complete rewrite. Some things can probably also be removed like the html (definitely) and tex (maybe) parsers.